### PR TITLE
Doctors endpoint

### LINKED ISSUE #2: Doctors Endpoints - API - 2.5pt
Doctors endpoints

- [x] Create Doctor model
- [x] Migration should contain these columns: name, profile pic, specialty, age, bio, consultation_fee, experience
- [x] Doctor model validation
- [x] Create an /doctors endpoint which returns a json response with doctors
- [x] Create a /doctors POST endpoint to create new doctor
- [x] Create a /doctors/id endpoint to get a doctors by id
- [x] Create a /doctors/id DELETE endpoint
- [x] Spec

### PR DESCRIPTION
**Pull request by codepantha**
## Changes introduced via this PR 👇 
- [x] Create Doctor model
- [x] Migration should contain these columns: name, profile pic, specialty, age, bio, consultation_fee, experience
- [x] Doctor model validation
- [x] Create an /doctors endpoint which returns a json response with doctors
- [x] Create a /doctors POST endpoint to create new doctor
- [x] Create a /doctors/id endpoint to get a doctors by id
- [x] Create a /doctors/id DELETE endpoint
- [x] Spec

### CODE REVIEW

File: app/controllers/api/V1/doctors_controller.rb [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  def create
+    doctor = Doctor.new(doctor_params)
+    doctor.age = age
+
+    if doctor.save
+      render json: doctor, status: 201
+    else
+      render json: { error: doctor.errors }, status: 422
+    end
+  end
```
Comment (AlexRS90) [inline app/controllers/api/V1/doctors_controller.rb:25]: Here we have a bug, when I tried to add a new doctor from postman is showing the following response: ⬇️  ⬇️ 

Captura de Pantalla 2022-03-28 a la(s) 8 58 39

Kindly fix it to have a good API performance 🤓

File: app/controllers/api/V1/doctors_controller.rb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+  def destroy
+    if @doctor.blank?
+      render json: { error: "Doctor with id #{params[:id]} not found" }, status: 404
+    else
+      @doctor.delete
+      render status: 204
+    end
+  end
```
Comment (AlexRS90) [inline app/controllers/api/V1/doctors_controller.rb:34]: With the delete action, we have a bug as well, in this case, is about cors permission, so I believe it just needs to give it the right permissions to allow the action, here is the API response:

Captura de Pantalla 2022-03-28 a la(s) 9 07 15

Kindly give it the right cors permissions 🤓  ✅

File: spec/models/doctor_spec.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+require 'rails_helper'
+
+RSpec.describe Doctor, type: :model do
+  pending "add some examples to (or delete) #{__FILE__}"
+end
```
Comment (AlexRS90) [inline spec/models/doctor_spec.rb]: I believe is a good idea to delete files if you are not using it, I noticed you have your specs in the requests folder and all are passing, that's great but keeping this empty  file spec the terminal is showing this warning: 🔽  🔽 
Captura de Pantalla 2022-03-28 a la(s) 9 21 53

Kindly remove the file to keep your files structure clean! 🤓

**Review by AlexRS90 — CHANGES_REQUESTED** [review]
# Hello @codepantherr 
### What an amazing job so far, you are implementing almost all the requirements for this milestone but there are some minor bugs to fix. 🐛  ⚠️ 

## Highlights: 

✅  Profesional repository
✅  Descriptive PR
✅  Descriptive commit messages
✅  No linters errors

To see the requested changes please scroll down to the comments

You're almost there keep rocking!! 🎸  🚀

File: app/controllers/api/V1/doctors_controller.rb (hunk 1 above)
Comment (codepantha) [reply app/controllers/api/V1/doctors_controller.rb:25]: Hi @AlexRS90 👋,

Thank you for your review. I noticed that you are trying to create a new resource using the params tab. I would suggest you create the fields as JSON.

File: app/controllers/api/V1/doctors_controller.rb (hunk 2 above)
Comment (codepantha) [reply app/controllers/api/V1/doctors_controller.rb:34]: Thank you. I'll fix this CORS issue immediately.

File: app/controllers/api/V1/doctors_controller.rb (hunk 1 above)
Comment (AlexRS90) [reply app/controllers/api/V1/doctors_controller.rb:25]: Hello @codepantherr 👋,

You're right, I can post a new doctor, this issue is solved!

File: spec/models/doctor_spec.rb (hunk 3 above)
Comment (codepantha) [reply spec/models/doctor_spec.rb]: Done ✔️

File: config/initializers/cors.rb [hunk 4]
```diff
@@ -5,12 +5,12 @@
...
-#     resource "*",
-#       headers: :any,
-#       methods: [:get, :post, :put, :patch, :delete, :options, :head]
-#   end
-# end
+    resource "*",
+      headers: :any,
+      methods: [:get, :post, :put, :patch, :delete, :options, :head]
+  end
+end
```
Comment (AlexRS90) [inline config/initializers/cors.rb:16]: Nice approach to solving the requested issue! 👨‍💻

**Review by AlexRS90 — APPROVED** [review]
## Hello @codepantherr 
### Your project is complete, feel free to merge it!
## STATUS APPROVED 🚀  🟢  🚀  🟢

File: config/initializers/cors.rb (hunk 4 above)
Comment (codepantha) [reply config/initializers/cors.rb:16]: @AlexRS90 thank you 🙏

**Pull request merged**